shared/contracts/BaseTemplate.sol

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /* VAULT */
+
+    function _installVaultApp(Kernel dao) internal returns (Vault) {
+        Vault vault = Vault(_installDefaultApp(dao, VAULT_APP_ID));
+        vault.initialize();
+        return vault;
+    }
+
+    function _createVaultPermissions(ACL acl, Vault vault, address grantee, address manager) internal {
+        acl.createPermission(grantee, vault, vault.TRANSFER_ROLE(), manager);
+    }
+
+    /* VOTING */
+
+    function _installVotingApp(Kernel dao, MiniMeToken token, uint64 support, uint64 acceptance, uint64 duration) internal returns (Voting) {
+        Voting voting = Voting(_installNonDefaultApp(dao, VOTING_APP_ID));
+        voting.initialize(token, support, acceptance, duration);
+        return voting;
+    }
+
+    function _createVotingPermissions(ACL acl, Voting voting, address grantee, address manager) internal {
+        acl.createPermission(grantee, voting, voting.MODIFY_QUORUM_ROLE(), manager);
+        acl.createPermission(grantee, voting, voting.MODIFY_SUPPORT_ROLE(), manager);
+    }
+
+    /* EVM SCRIPTS */
+
+    function _createEvmScriptsRegistryPermissions(ACL acl, address grantee, address manager) internal {
+        EVMScriptRegistry registry = EVMScriptRegistry(acl.getEVMScriptRegistry());
+        acl.createPermission(grantee, registry, registry.REGISTRY_MANAGER_ROLE(), manager);
+        acl.createPermission(grantee, registry, registry.REGISTRY_ADD_EXECUTOR_ROLE(), manager);
+    }
+
+    /* APPS */
+
+    function _installNonDefaultApp(Kernel dao, bytes32 appId) internal returns (address) {
+        return _installApp(dao, appId, new bytes(0), false);
+    }
+
+    function _installDefaultApp(Kernel dao, bytes32 appId) internal returns (address) {
+        return _installApp(dao, appId, new bytes(0), true);
+    }
+
+    function _installApp(Kernel dao, bytes32 appId, bytes data, bool setDefault) internal returns (address) {
+        address latestBaseAppAddress = _latestVersionAppBase(appId);
+        address instance = address(dao.newAppInstance(appId, latestBaseAppAddress, data, setDefault));
+        emit InstalledApp(instance, appId);
+        return instance;
+    }
+
+    function _latestVersionAppBase(bytes32 appId) internal view returns (address base) {
+        Repo repo = Repo(PublicResolver(ens.resolver(appId)).addr(appId));
+        (,base,) = repo.getLatest();
+    }
+
+    /* TOKEN */
+
+    function _createToken(string name, string symbol) internal returns (MiniMeToken) {
+        require(isContract(address(miniMeFactory)), "TEMPLATE_MINIME_FACTORY_NOT_PROVIDED");
+        MiniMeToken token = miniMeFactory.createCloneToken(MiniMeToken(address(0)), 0, name, 18, symbol, true);
+        emit DeployToken(address(token));
+        return token;
+    }
+
+    /* IDS */
+
+    function _registerID(string name, address owner) internal {
+        require(isContract(address(aragonID)), "TEMPLATE_ARAGON_ID_NOT_PROVIDED");
```

Comment:
Since both the `registerID()` and `createToken()` will likely get called a lot, we can optimize their gas by changing the require to `require(address(aragonID) !== address(0), ERROR_CONSTANT)` and only checking if it's a contract in the constructor.